src/pixelator/plot/__init__.py

```diff
@@ -935,6 +939,165 @@ def plot_colocalization_diff_volcano(
+
+    This function creates a scatter plot of abundance data for two markers colored
+    based on region density:
+    Example usage: `density_scatter_plot(pxl.adata, "CD3E", "CD4")`.
+    It is also possible to specify different conditions for faceting the plot by
+    columns and rows:
+    Example usage: `density_scatter_plot(pxl.adata, "CD3E", "CD4",
+                        facet_row = "stimulation", facet_column = "donor")`.
+    facet_row and facet_column should be names of categorical columns in adata.obs.
+    In addition, a gate can be specified as a Series with xmin, xmax, ymin, and ymax
+    to mark a range for components of interest. Alternatively, gate can be specified
+    as a DataFrame to allow for different gate ranges for various conditions.
+    When both facet_row and facet_column are specified, the condition becomes a
+    tuple (facet_row, facet_column), if only one is specified, that parameter
+    becomes the condition. The condition permuations are used as the index of
+     the gate.
+    Example usage:
+        gate = pd.DataFrame(columns = ["xmin", "ymin", "xmax", "ymax"])
+        gate.loc["Resting"] = [2, 2, 5, 4]
+        gate.loc["PHA"] = [1.5, 1.5, 5, 4]
+        fig, ax = density_scatter_plot(pixel.adata, "CD3E", "CD4", layer="dsb",
+                    facet_column="sample", gate=gate)
+
+
+    :param adata: Anndata object containing the marker abundance data per
+     component.
+    :param marker1: The first marker to plot (x-axis).
+    :param marker2: The second marker to plot (y-axis).
+    :param layer: The anndata layer (e.g. transformation) to use for the marker
+     data. Defaults to None.
+    :param facet_row: The column to use for faceting the plot by rows.
+     Defaults to None.
+    :param facet_column: The column to use for faceting the plot by columns.
+     Defaults to None.
+    :param gate: The gate to use for marking a range of interest. Defaults to
+     None.
+    """
+    layer_data = adata.to_df(layer)
+    data = layer_data.loc[:, [marker1, marker2]]
+    data.loc[:, "density"] = gaussian_kde(data.T)(data.T)
+    if facet_column is not None:
+        data.loc[:, facet_column] = adata.obs.loc[:, facet_column]
+
+    if facet_row is not None:
+        data.loc[:, facet_row] = adata.obs.loc[:, facet_row]
+
+    if facet_column is not None or facet_row is not None:
+        plot_grid = sns.FacetGrid(data=data, col=facet_column, row=facet_row)
+        plot_grid.map_dataframe(
+            sns.scatterplot,
+            x=marker1,
+            y=marker2,
+            hue="density",
+            palette=jet_colormap,
+            size=0.1,
+        )
+        if gate is not None:
+            plot_grid.map_dataframe(
+                __add_gate_box,
+                gate=gate,
+                marker1=marker1,
+                marker2=marker2,
+                facet_row=facet_row,
+                facet_column=facet_column,
+            )
+    else:
```

Comment:
Is it possible to remove marginal densities when plotting a single facet? I think some users might not want to use them every time

Reply:
Done, show_marginal is now an input and defaults to false.